src/main/java/uk/gov/hmcts/reform/prl/courtnav/mappers/FL401ApplicationMapper.java

```diff
@@ -217,17 +227,58 @@ public CaseData mapCourtNavData(CourtNavFl401 courtNavCaseData) throws NotFoundE
             .caseSubmittedTimeStamp(DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(zonedDateTime))
             .build();
 
-        caseData = caseData.toBuilder()
-            .courtName(getCourtName(caseData))
-            .courtEmailAddress(getCourtEmailAddress(court))
-            .build();
-
+        caseData = populateCourtDetailsForCourtNaveCase(authorization, caseData,
+                                                        courtNavCaseData.getMetaData().getCourtSpecialRequirements());
         caseData = caseData.setDateSubmittedDate();
 
         return caseData;
 
     }
 
+    private CaseData populateCourtDetailsForCourtNaveCase(String authorization, CaseData caseData,
+                                                          String epimsId) throws NotFoundException {
+        Optional<CourtVenue> courtVenue = Optional.empty();
+        //1. get court details from provided epimsId request
+        if (!StringUtils.isEmpty(epimsId)) {
+            courtVenue = getCourtVenue(authorization, epimsId);
+        }
+        //2. if not found check launch-darkly flag and populate default Swansea court Id.
+        if (launchDarklyClient.isFeatureEnabled(PrlLaunchDarklyFlagConstants.COURTNAV_SWANSEA_COURT_MAPPING)
+            && courtVenue.isEmpty()) {
+            epimsId = COURTNAV_DUMMY_BASE_LOCATION_ID;
```

Comment:
I think we should give epimsId of swansea?
@AlokDatta , what do you say.

Reply:
@yogendraU this COURTNAV_DUMMY_BASE_LOCATION_ID contains epims id of swansea court.
cc @AlokDatta